assets/src/hooks/v2/use_interval_paging.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect, useRef, useCallback } from "react";
+
+interface UseIntervalPagingProps {
+  numPages: number; // Total number of pages to cycle through
+  cycleIntervalMs?: number; // In milliseconds, the interval for cycling pages (only used if advanceOnDataRefresh is false)
+  onFinish: () => void; // Callback when cycling completes
+}
+
+/**
+ * This hook acts as a way for components with multiple pages to advance
+ * through their pages on a set time interval.
+ */
+function useIntervalPaging({
+  numPages,
+  cycleIntervalMs,
+  onFinish,
+}: UseIntervalPagingProps) {
+  const [pageIndex, setPageIndex] = useState(0);
+  // Use refs to keep stable references to numPages and interval without triggering re-renders
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const numPagesRef = useRef(numPages);
+
+  useEffect(() => {
+    numPagesRef.current = numPages;
+  }, [numPages]);
+
+  // Callback that handles changing the state of pageIndex
+  const advancePage = useCallback(() => {
+    setPageIndex((prevIndex) => {
+      return (prevIndex + 1) % numPagesRef.current;
+    });
+  }, []);
+
+  // Start the interval for time-based advancement
+  const startTimer = useCallback(() => {
+    if (intervalRef.current) clearInterval(intervalRef.current); // Clear any existing timer
```

Comment:
As long as we return a cleanup function from `useEffect` below, the interval should get cleared any time the effect needs to be re-run, so we shouldn't need to do that here as well.

Reply:
This is overcomplicated from when I had the interval and refresh hooks combined and more logic was required. Good point that this shouldn't be needed anymore